COMMUNICATIONS.md

```diff
@@ -1,3 +1,199 @@
+* Encourage a healthy form of discussion, avoiding a situation where
+  we're afraid to have any technical disagreement
+
+## Prior art
+
+Some related documents I've written in the past are tangentially
+related:
+
+* [Effective Ways to Get Help from Maintainers](https://www.snoyman.com/blog/2017/10/effective-ways-help-from-maintainers)
+* [How to send me a pull request](https://www.snoyman.com/blog/2017/06/how-to-send-me-a-pull-request)
+* [Maintainer communication](https://github.com/commercialhaskell/commercialhaskell/blob/master/guide/maintainer-communication.md)
+* [Guide to open source maintenance](https://github.com/commercialhaskell/commercialhaskell/blob/master/guide/open-source-maintenance.md)
+
+## Principles
+
+The following are principles I recommend. If you disagree with these,
+would like to propose changes to the wording, or would like to propose
+others, feel free to open an issue or pull request on this repository.
+
+### Don't take it personally
+
+We often become personally invested in our software. We dedicate huge
+amounts of time, both personal and professional, to making it
+succeed. It can sometimes be difficult to hear criticism of our
+software.
+
+However, technical discussions will ultimately include some such
+criticism. Don't take it personally! Many of the points below try to
+help this become a reality.
+
+### Don't make it personal
+
+A corollary to the above: don't _make_ it personal. When discussing a
+technical topic, there is rare&mdash;if ever&mdash;good reason to
+implicate negative attributes to the author. Indicating that the
+author is unintelligent, implying nefarious motives, or similar
+comments brings the discussion down from technical to personal.
+
+_If_ the other side in a discussion makes it personal, my
+recommendation: immediately point it out as such, with a link to this
+section.
+
+### Respect privacy
+
+Some topics are best discussed publicly. Others may be well handled
+privately. If you believe you are in the latter category, send a
+private message requesting the other side participate privately. If
+agreed to, somewhat obviously, **do not** violate that agreement!
+
+That said, private communications can be used as a method of private
+attacks and intimidation. As such, communications can only reasonably
+be considered private if both sides agree.
+
+### Avoid ad hominems
+
+Do not discredit someone's technical work or discussions based on who
+made the comment or did the work. This can come in many different
+forms:
+
+* **BAD** Their previous work is shoddy, so this work must be shoddy.
+* **BAD** They are part of an organization/company/committee,
+  therefore they are inherently biased and cannot be listened to.
+* **BAD** They are seen as friendly with someone else, therefore they
+  are inherently biased.
+
+A related topic is the pattern of behavior argument. "Person X has
+done A, B, and C, therefore D must have a similar motivation." That's
```

Comment:
```suggestion
done A, B, and C, therefore Y must have a similar motivation." That's
```

Reply:
I actually meant something slightly different, I'll include a change to see if it clarifies.

Reply:
I ended up with this, how's that?

```diff
-done A, B, and C, therefore D must have a similar motivation." That's
+done A, B, and C, therefore when X does D it must have a similar motivation." That's
```

Reply:
Crystal clear. I had misunderstood it indeed; this dispels my confusion.